validator/src/bootstrap.rs

```diff
@@ -1210,33 +1210,37 @@ mod with_incremental_snapshots {
                 .any(|hay| needle.0 == hay.0 && needle.1 != hay.1)
         }
 
-        'outer: for node in nodes {
+        'to_next_node: for node in nodes {
             // First get the full snapshot hashes for each node and add them as the keys in the
             // known snapshot hashes map.
             let full_snapshot_hashes = get_full_snapshot_hashes_for_node(node);
-            for full_snapshot_hash in &full_snapshot_hashes {
+            '_to_next_full_snapshot: for full_snapshot_hash in &full_snapshot_hashes {
```

Comment:
Very gnarly indeed. Each time I've looked at this function I've tried ways to make it simpler. Hopefully one day!

As for the unreferenced label, yeah... I did this because I have labels on the other for loops, so I decided that maybe this would be easier to read, knowing that all the loops have a label (when I'm reading code, inconsistencies give me pause (i.e. "I know a label is not used here, but is that right? Was it refactored/overlooked?"), but the same thing can be said for an unused label...).

If this is the only change to the PR, maybe I can defer the change (as to not go through CI again)? And if there are other changes, then I'll go ahead and remove the label.

Reply:
Very gnarly indeed. Each time I've looked at this function I've tried ways to make it simpler. Hopefully one day!

As for the unreferenced label, yeah... I did this because I have labels on the other for loops, so I decided that maybe this would be easier to read, knowing that all the loops have a label (when I'm reading code, inconsistencies give me pause (i.e. "I know a label is not used here, but is that right? Was it refactored/overlooked?"), but the same thing can be said for an unused label...).

If this is the only change to the PR, maybe I can defer the change (as to not go through CI again)? And if there are other changes, then I'll go ahead and remove the label.

Reply:
sgtm